lib/std/net.zig

```diff
@@ -1728,6 +1728,16 @@ pub const StreamServer = struct {
 
         const sockfd = try os.socket(address.any.family, sock_flags, proto);
         self.sockfd = sockfd;
+
+        if (std.io.is_async and builtin.os.tag == .windows) {
+            _ = os.windows.CreateIoCompletionPort(
+                sockfd,
+                std.event.Loop.instance.?.os_data.io_port,
+                0,
+                0,
+            ) catch undefined;
```

Comment:
Should this be moved into `os.socket` given this happens in `os.accept`?

Reply:
At first I did place it there, but it seemed out of place. It's possible that I misunderstood the purpose of os file. At the end I moved it to net to allow a higher granularity of control -> so that someone could create socket without connecting it to the std iocp but I'm not exactly sure if that's a valid case.

Reply:
For consistency, I think the CreateIoCompletionPort() to the event loop should happen in `std.net` instead of `std.os` as net does `is_async`/`Loop` interactions already while os only deals with the socket itself (not the event Loop) even for things like `read/write/send/recv`